pyls/plugins/importmagic_lint.py

```diff
@@ -0,0 +1,264 @@
+# Copyright 2017 Palantir Technologies, Inc.
+import logging
+import re
+import sys
+import importmagic
+from pyls import hookimpl, lsp, _utils
+
+
+log = logging.getLogger(__name__)
+
+SOURCE = 'importmagic'
+ADD_IMPORT_COMMAND = 'importmagic.addimport'
+REMOVE_IMPORT_COMMAND = 'importmagic.removeimport'
+MAX_COMMANDS = 4
+UNRES_RE = re.compile(r"Unresolved import '(?P<unresolved>[\w.]+)'")
+UNREF_RE = re.compile(r"Unreferenced import '(?P<unreferenced>[\w.]+)'")
+
+_index_cache = {}
+
+
+def _get_index(sys_path):
```

Comment:
Let's make this return a future, that way we can skip results instead of blocking on indexing.